addons/point_of_sale/models/account_bank_statement.py

```diff
@@ -10,6 +11,12 @@ class AccountBankStatement(models.Model):
     pos_session_id = fields.Many2one('pos.session', string="Session", copy=False)
     account_id = fields.Many2one('account.account', related='journal_id.default_debit_account_id', readonly=True)
 
+    @api.multi
+    def check_confirm_bank(self):
+        if self.state  == 'open':
```

Comment:
self is a bank statement, so I don't think that's what you want to do, as you won't be able to confirm any open bank statement

Reply:
for bs in self:
   if bs.pos_session_id.state == 'open':
       raise